src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py

```diff
@@ -241,14 +308,20 @@ def _on_rpc_done(
             if future.code() == grpc.StatusCode.OK:
                 logger.debug("Successful response.")
             else:
-                logger.debug(f"RPC failed: {call}")
+                logger.debug(f"RPC failed: {rpc_id}")
     with _global_lock:
         for watcher in _watchers:
-            watcher.on_rpc_complete(rpc_id, hostname, method)
+            watcher.on_rpc_complete(
+                rpc_id,
+                hostname,
+                method,
+                future.initial_metadata(),
+                future.trailing_metadata(),
+            )
 
 
 def _remove_completed_rpcs(
-    futures: Mapping[int, grpc.Future], print_response: bool
+    futures: Mapping[int, FutureFromCallType], print_response: bool
```

Comment:
Oh, this shadows the `futures` import. It'll be better to rename this argument too.

Reply:
Changed this to `rpc_futures`.